### DISCUSSION TITLE
Clock source being used in NucleoF103RB

### DISCUSSION THREAD

**Discussion by agganitk** [board707/DMD_STM32#98] [General]
Hello,
Can you tell me what clock source is being used to run the code? Is it Internal HSI or External coming from ST-LINK MCU ? I am much familier with STMCUBEIDE . I tried to go through code and ARDUINO IDE but can not find. 
I having issue if I use external power supply to power the nucelo board then the code doesn't run until reset button is pressed once.

**Comment by board707** [comment]
> Can you tell me what clock source is being used to run the code? Is it Internal HSI or External coming from ST-LINK MCU ?

Hi
Because there is no external St-Link on used STM32F103 "Bluepill" board, I would think that it is an internal HSI

**Comment by agganitk** [comment]
Hi..I m using NUCLEOF103RB .. it has external ST_LINK.    
And as I said I have to press reset button once on giving external power supply to make the code work properly. If I give power via USB then I do not need to press reset button.  I want to check clock selection code. I checked Libraries but can not find where RCC clock and registeres are declared..

**Reply by board707** [reply]
> I checked Libraries but can not find where RCC clock and registeres are declared..

I think it done in the Clark's STM32 package. For `generic_stm32f103c` variant, see the files `board.cpp` and `board_setup.cpp` in 
`STM32F1/variants/generic_stm32f103c/wirish`,
